packages/terra-button/src/variants/_AnchorButton.jsx

```diff
@@ -0,0 +1,267 @@
+    // Add active state to FF browsers
+    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE) {
+      this.setState({ active: true });
+
+      // Follow href on space keydown when rendered as an anchor tag
+      if (this.props.href) {
+        // Prevent window scrolling
+        event.preventDefault();
+        window.location.href = this.props.href;
+      }
+    }
+
+    // Add focus styles for keyboard navigation
+    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
+      this.setState({ focused: true });
+    }
+
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(event);
+    }
+  }
+
+  handleKeyUp(event) {
+    // Remove active state from FF broswers
+    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE) {
+      this.setState({ active: false });
+    }
+
+    // Apply focus styles for keyboard navigation
+    if (event.nativeEvent.keyCode === KeyCode.KEY_TAB) {
+      this.setState({ focused: true });
+    }
+
+    if (this.props.onKeyUp) {
+      this.props.onKeyUp(event);
+    }
+  }
+
+  handleFocus(event) {
+    if (this.shouldShowFocus) {
+      this.setState({ focused: true });
+    }
+
+    if (this.props.onFocus) {
+      this.props.onFocus(event);
+    }
+  }
+
+  handleMouseDown(event) {
+    if (this.props.onMouseDown) {
+      this.props.onMouseDown(event);
+    }
+
+    // See https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/focus#Notes
+    // If you call HTMLElement.focus() from a mousedown event handler, you must call event.preventDefault() to keep the focus from leaving the HTMLElement.
+    // Otherwise, when you click on the button again, focus would leave the button and onBlur would get called causing the document.activeElement would no longer be the focused button.
+    event.preventDefault();
+  }
+
+  render() {
+    const {
+      isBlock,
+      isCompact,
+      isDisabled,
+      text,
+      type,
+      variant,
+      href,
+      onClick,
+      onMouseDown,
+      onBlur,
+      onFocus,
+      onKeyDown,
+      onKeyUp,
+      refCallback,
+      title,
+      ...customProps
+    } = this.props;
+
+    const theme = this.context;
+
+    const buttonClasses = classNames(
+      cx([
+        'button',
+        variant,
+        { 'is-disabled': isDisabled },
+        { block: isBlock },
+        { compact: isCompact },
+        { 'is-active': this.state.active && !isDisabled },
+        { 'is-focused': this.state.focused && !isDisabled },
+        theme.className,
+      ]),
+      customProps.className,
+    );
+
+    const buttonLabelClasses = cx([
+      'button-label',
+      'text-only',
+    ]);
+
+    const ariaLabel = customProps['aria-label'];
+
+    return (
+      <a
+        {...customProps}
+        className={buttonClasses}
+        type={type}
+        tabIndex={isDisabled ? '-1' : customProps.tabIndex}
+        aria-disabled={isDisabled}
+        aria-label={ariaLabel || text}
```

Comment:
I two other a11y questions: 

1. Why do we also have `aria-label` as well as `text`? Since text is required then would it not be sufficient to leave out aria-label?
2. What scenario would a consumer set `aria-disabled` yet not also set the `disabled` attr of the link? My guess would be if they wanted custom styling for a disabled button. Is that true?

Reply:
1. `aria-label` could be used by consumers to provide additional semantics along with the regular visible `text`. 
2. They can choose to disable it using `isDisabled` prop (similar to terra-button).